### PR TITLE
IDE0063 code fix now keeps nearby comments

### LINKED ISSUE #48586: IDE0063 code fixer deletes comments nearby using statement or curly braces
**Version Used**: Visual Studio 2019 Version 16.7.5

**Steps to Reproduce**: 
1.  Write a following code, which has an `IDE0063 'using' statement can be simplified`:
```cs
using System;
using System.IO;

class C
{
    static void Main()
    {
        using (var s = new MemoryStream()) // foo
        { // bar
            Console.WriteLine(s.CanRead); // Only this comment remains.
        } // baz
    }
}
```
2. Apply  the `Use simple 'using' statement`

**Expected Behavior**:
All comments remain like this:
```cs
using System;
using System.IO;

class C
{
    static void Main()
    {
        using var s = new MemoryStream(); // foo
        // bar
        Console.WriteLine(s.CanRead); // Only this comment remains.
        // baz
    }
}
```

**Actual Behavior**:
"foo", "bar", "baz" comments are deleted:
```cs
using System;
using System.IO;

class C
{
    static void Main()
    {
        using var s = new MemoryStream();
        Console.WriteLine(s.CanRead); // Only this comment remains.
    }
}
```

### PR DESCRIPTION
**Pull request by louis-z**
Fixes #48586

### CODE REVIEW

File: src/Analyzers/CSharp/CodeFixes/UseSimpleUsingStatement/UseSimpleUsingStatementCodeFixProvider.cs [hunk 1]
```diff
@@ -118,9 +118,27 @@ private static SyntaxTriviaList Expand(List<StatementSyntax> result, UsingStatem
...
             {
                 case BlockSyntax blockSyntax:
+                    var statements = blockSyntax.Statements;
+
+                    var openBraceTrailingTrivia = blockSyntax.OpenBraceToken.TrailingTrivia;
+                    if (openBraceTrailingTrivia.Any(t => t.IsSingleOrMultiLineComment()))
+                    {
+                        var firstStatement = statements.First()
+                            .WithPrependedLeadingTrivia(openBraceTrailingTrivia);
+                        statements = new SyntaxList<StatementSyntax>(statements.Skip(1).Prepend(firstStatement));
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseSimpleUsingStatement/UseSimpleUsingStatementCodeFixProvider.cs:128]:
```suggestion
                        statements = statements.Replace(statementsFirst, firstStatement);
```

File: src/Analyzers/CSharp/CodeFixes/UseSimpleUsingStatement/UseSimpleUsingStatementCodeFixProvider.cs [hunk 2]
```diff
@@ -118,9 +118,27 @@ private static SyntaxTriviaList Expand(List<StatementSyntax> result, UsingStatem
...
+                    }
+
+                    var closeBraceTrailingTrivia = blockSyntax.CloseBraceToken.TrailingTrivia;
+                    if (closeBraceTrailingTrivia.Any(t => t.IsSingleOrMultiLineComment()))
+                    {
+                        var lastStatement = statements.Last()
+                            .WithAppendedTrailingTrivia(closeBraceTrailingTrivia);
+                        statements = new SyntaxList<StatementSyntax>(statements.Take(statements.Count - 1).Append(lastStatement));
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseSimpleUsingStatement/UseSimpleUsingStatementCodeFixProvider.cs:136]: update similarly.

**Review by CyrusNajmabadi — APPROVED** [review]
LGTM with suggested changes.

File: src/Analyzers/CSharp/Tests/UseSimpleUsingStatement/UseSimpleUsingStatementTests.cs [hunk 3]
```diff
@@ -1384,6 +1384,86 @@ static void Main()
...
+
+class C
+{
+    void M()
+    {
+        // Make...
+        [||]using (var a = b) // ...sure...
+        { // ...that...
+            Console.WriteLine(s.CanRead); // ...all...
+        } // ...comments...
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/UseSimpleUsingStatement/UseSimpleUsingStatementTests.cs:1405]: Can you add a basic test that only has the comments on `{` and `}`.  thanks :)

**Comment by louis-z** [comment]
Hey @CyrusNajmabadi,
AFAICT everything was resolved, but the last build isn't green. :(

**Comment by CyrusNajmabadi** [comment]
we're having some CI stability problems.  So i kicked off hte tests again :)

**Comment by louis-z** [comment]
Hey @CyrusNajmabadi,
Just wanted to say the last build was green. ;)

**Comment by CyrusNajmabadi** [comment]
Booya! :)  Thanks for this.

**Pull request merged**